### ISSUE TITLE
"wasm-ld: error: symbol exported via --export not found: _get_environ" when compiling and linking separately

### ISSUE THREAD

**Issue by roepel** [emscripten-core/emscripten#10579]
I am currently working on a WASM Application that uses SDL2, SDL2_image and SDL2_mixer.
When building the project with the one-line command
```
emcc -std=c++11 -I [..] -s USE_SDL=2 -s USE_SDL_IMAGE=2 -s USE_SDL_MIXER=2 -s WASM=1 ../client.cpp -o client.js
```
everything works just fine.
However, when building using two separate commands for compiling to .o and then linking, which look like these
```
emcc -std=c++11 -I [..] -s USE_SDL=2 -s USE_SDL_IMAGE=2 -s USE_SDL_MIXER=2 -s WASM=1 -o client.o  -c ../client.cpp
```
and then linking with
```
emcc -s LLD_REPORT_UNDEFINED -s USE_SDL=2 -s USE_SDL_IMAGE=2 -s USE_SDL_MIXER=2 client.o -o client.js
```
an error appears that says `wasm-ld: error: symbol exported via --export not found: _get_environ`.

Am I missing something or is this a bug?

**Comment by sbc100** [comment]
`_get_environ` should be defined in `libc.a` (it comes from `system/lib/libc/extras.c`).

Can you run the failing command with `-v` to see the linker command in full and post it here.

Then you can try running `llvm-nm` on the `libc.a` thats being using.  You should see extras.o with that symbol in it.

I see that you are using the `-s LLD_REPORT_UNDEFINED`.. this flag is still experimental and doesn't always give accurate results.   Does the link work without that?    I will add a warning about LLD_REPORT_UNDEFINED to avoid people relying on it until it works universally.

**Comment by roepel** [comment]
Yes, `-s LLD_REPORT_UNDEFINED` did cause this problem.
I used this flag since I did not know/forgot to pass the USE_SDL* stuff to the linker in CMakeLists.txt with target_link_options(). That gave some undefined symbol errors, with the suggestion to use `-s LLD_REPORT_UNDEFINED`.
It did not help much, but I did't bother to remove it when calling emcc myself to debug the problem. However if I do not use this flag, the linking stage does not fail and I can compile the project using CMake.
